### ISSUE TITLE
Missing pretrained SGN attack models in repository 

### ISSUE THREAD

**Issue by sihatafnan** [Thomasc33/Privacy-Retargeting#2]
The evaluation script (legacy_scripts/eval.py) requires three pretrained SGN model files that are not included in the repository:

- SGN/pretrained/privacy_60_sgnpt.pt — actor re-identification attack model
- SGN/pretrained/action_60_sgnpt_no_two_actor.pt — action recognition utility model
- SGN/pretrained/gender.pt — gender classification attack model

Without these files, it is not possible to reproduce the evaluation metrics reported in Table 1 of the paper (re-identification accuracy, action recognition accuracy, gender classification accuracy).
Could you make these pretrained models available, either by:

- Adding them to the repository directly
- Uploading them to Google Drive (similar to the NTU skeleton data) and linking them in the README
- Or any other hosting method you prefer

Thank you for the great work on the paper!

**Comment by sihatafnan** [comment]
Resolved — SGN pretrained models can be trained from the existing code
The three pretrained SGN attack models (privacy_60_sgnpt.pt, action_60_sgnpt_no_two_actor.pt, gender.pt) are not included in the repository but can be reproduced using the existing SGN/train.py script with the following steps.
Prerequisites: Complete the preprocessing pipeline to generate NTU/SGN/X_full.pkl as described in the README. Note two bugs in seq_transformation.py must be fixed first (see PR #[your PR number] or apply manually):

Replace np.string_ with np.bytes_ (NumPy 2.0 compatibility)
Comment out the align_frames() call (causes OOM kill due to ~8.6GB single allocation)

Fix data format in SGN/train.py: The script expects 3D arrays but X_full.pkl stores flattened 2D arrays. In load_and_split_cv, replace:
```
pythondata = data[:, :, :3]  # keep only xyz
```
with:
```
pythonif data.ndim == 3:
    data = data[:, :, :3]
elif data.ndim == 2:
    data = data[:, :75].reshape(-1, 25, 3)
```
Train the re-identification model:
```
mkdir -p SGN/pretrained
python SGN/train.py \
    --task privacy \
    --dataset ntu60 \
    --data-path NTU/SGN/X_full.pkl \
    --output SGN/pretrained/privacy_60_sgnpt.pt
```
Train the action recognition model:
```
bashpython SGN/train.py \
    --task action \
    --dataset ntu60 \
    --data-path NTU/SGN/X_full.pkl \
    --output SGN/pretrained/action_60_sgnpt_no_two_actor.pt
```
Gender model: SGN/train.py does not currently support gender classification training (--task gender). This would require adding a gender label parser and a 2-class output head.
Reproduced results (120 epochs, NTU-60 CV split):

Re-ID Top-1: 77.1% | Top-5: 97.0% (paper: 87.8% / 97.3%)
The Top-5 match confirms correct learning. Top-1 gap is likely due to two-actor actions not being filtered and no data augmentation.
